test/native/vitest.config.ts

```diff
@@ -0,0 +1,28 @@
+import * as path from 'node:path'
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
```

Comment:
This seems to be the exact copy of the existing vitest config for node tests. Cannot we just reference to it for native tests, which I assume is the intention here?

Reply:
yes we probably can.  I was separating this to try and get a test environment that worked properly for `react-native` but failed to do so, and kept this to provide a path to doing that in the future.  I think it's probably ok to duplicate here, but we could also nest the `native` test inside of the node folder and just run it there, since it's fairly small. 

No strong preferences, although a working native setup would be ideal.  I couldn't even get `vi.mock` to throw for builtins here, which I also tried

Reply:
I made the `vi.mock()` work but it requires a module alias hack in Vitest.